File: moderate/try_to_solve_it.lua
function deco(c, o)
  return string.char(string.byte(c) + o)
end

for line in io.lines(arg[1]) do
  for i in line:gmatch(".") do
    io.write(i:find("[a-f]") and deco(i, 20) or
             i:find("[g-m]") and deco(i, 7) or
             i:find("[n-t]") and deco(i, -7) or
             i:find("[u-z]") and deco(i, -20) or i)
  end
  print()
end

File: moderate/try_to_solve_it.clj
(defn deco [a]
  (case a
    (\a \b \c \d \e \f) (char (+ (int a) 20))
    (\g \h \i \j \k \l \m) (char (+ (int a) 7))
    (\n \o \p \q \r \s \t) (char (- (int a) 7))
    (\u \v \w \x \y \z) (char (- (int a) 20))
    a))

(require '[clojure.string :as str])
(doseq [item (for [st (str/split-lines (slurp (first *command-line-args*)))]
  (map deco (seq st))
)] (println (str/join item)))
